lib/posthog/feature_flags.rb

```diff
@@ -118,33 +131,71 @@ def get_feature_flag(key, distinct_id, groups = {}, person_properties = {}, grou
     end
 
     def get_all_flags(distinct_id, groups = {}, person_properties = {}, group_properties = {}, only_evaluate_locally = false)
-      # returns a string hash of all flags
+    # returns a string hash of all flags
+      response = get_all_flags_and_payloads(distinct_id, groups, person_properties, group_properties, only_evaluate_locally)
+      flags = response[:featureFlags]
+    end
 
-      # make sure they're loaded on first run
+    def get_feature_payloads(distinct_id, groups = {}, person_properties = {}, group_properties = {}, only_evaluate_locally = false)
+      response = get_all_flags_and_payloads(distinct_id, groups, person_properties, group_properties, only_evaluate_locally)
+      payloads = response[:featureFlagPayloads]
+    end
+
+    def get_all_flags_and_payloads(distinct_id, groups = {}, person_properties = {}, group_properties = {}, only_evaluate_locally = false)
       load_feature_flags
 
-      response = {}
+      flags = {}
+      payloads = {}
       fallback_to_decide = @feature_flags.empty?
 
       @feature_flags.each do |flag|
         begin
-          response[flag[:key]] = _compute_flag_locally(flag, distinct_id, groups, person_properties, group_properties)
+          match_value = _compute_flag_locally(flag, distinct_id, groups, person_properties, group_properties)
+          flags[flag[:key]] = match_value
+          match_payload = _compute_flag_payload_locally(flag[:key], match_value)
+          if match_payload
+            payloads[flag[:key]] = match_payload
+          end
         rescue InconclusiveMatchError => e
           fallback_to_decide = true
         rescue StandardError => e
           logger.error "Error computing flag locally: #{e}."
           fallback_to_decide = true
         end
       end
-
       if fallback_to_decide && !only_evaluate_locally
         begin
-          flags = get_feature_variants(distinct_id, groups, person_properties, group_properties)
-          response = {**response, **flags}
+          flags_and_payloads = get_decide(distinct_id, groups, person_properties, group_properties)
+          symbolize_keys! flags
```

Comment:
this seems unnecessary?

Reply:
(this = symbolizing and then stringifying)

Reply:
It doesn't merge properly though if they're not all one type (either symbols or strings). I can convert them to all strings instead of symbolizing? 🤷‍♀️

Reply:
what's a string and what's not? This should be uniform 🤔  - anyway seems like a potential bug, would be nice to get to the bottom. I think we deal with everything being symbols, iirc, so everything coming should be a symbol.

Reply:
Actually you were already stringifying the keys here in `get_feature_variants` with the flags that come back from the decide endpoint 😮  

As we're setting the `flags` and `payloads` hash above, we have string key value pairs because the keys come from the feature flag `key` value which is a string, but the decide endpoint flags will return symbolized flags. It looks like you had made a decision previously to convert the flag results from the decide endpoint from symbols to strings instead so I'll just stick with that and then this avoids having to change all the test specs from strings to symbols if it's not needed